README.md

```diff
@@ -90,132 +90,138 @@ Single core benchmarks
-RangeQuery/at_modifier_                               27.3MB ± 0%    34.2MB ± 0%  +25.05%  (p=0.008 n=5+5)
-RangeQuery/at_modifier_with_positive_offset_vector    27.1MB ± 0%    34.0MB ± 0%  +25.23%  (p=0.016 n=4+5)
-RangeQuery/clamp                                      31.1MB ± 0%    26.3MB ± 0%  -15.56%  (p=0.008 n=5+5)
-RangeQuery/clamp_min                                  31.1MB ± 0%    26.2MB ± 0%  -15.72%  (p=0.008 n=5+5)
-RangeQuery/complex_func_query                         45.6MB ± 0%    29.2MB ± 0%  -35.94%  (p=0.008 n=5+5)
-RangeQuery/func_within_func_query                     32.1MB ± 0%    27.7MB ± 0%  -13.62%  (p=0.008 n=5+5)
-RangeQuery/aggr_within_func_query                     32.1MB ± 0%    27.7MB ± 0%  -13.61%  (p=0.008 n=5+5)
+RangeQuery/vector_selector                            19.5MB ± 0%    29.1MB ± 0%  +49.19%  (p=0.008 n=5+5)
+RangeQuery/sum                                        8.25MB ± 0%   10.28MB ± 0%  +24.59%  (p=0.008 n=5+5)
+RangeQuery/sum_by_pod                                 90.6MB ± 0%    19.4MB ± 0%  -78.60%  (p=0.008 n=5+5)
+RangeQuery/rate                                       20.5MB ± 0%    30.6MB ± 0%  +49.22%  (p=0.008 n=5+5)
+RangeQuery/sum_rate                                   9.31MB ± 0%   11.87MB ± 0%  +27.51%  (p=0.008 n=5+5)
+RangeQuery/sum_by_rate                                83.6MB ± 0%    20.9MB ± 0%  -75.04%  (p=0.008 n=5+5)
+RangeQuery/binary_operation_with_one_to_one           15.2MB ± 0%    14.9MB ± 0%   -2.43%  (p=0.008 n=5+5)
+RangeQuery/binary_operation_with_many_to_one          67.2MB ± 0%    35.4MB ± 0%  -47.37%  (p=0.008 n=5+5)
+RangeQuery/binary_operation_with_vector_and_scalar    34.1MB ± 0%    30.9MB ± 0%   -9.22%  (p=0.008 n=5+5)
+RangeQuery/unary_negation                             20.5MB ± 0%    30.1MB ± 0%  +46.77%  (p=0.008 n=5+5)
+RangeQuery/vector_and_scalar_comparison               32.7MB ± 0%    30.9MB ± 0%   -5.48%  (p=0.016 n=5+4)
+RangeQuery/positive_offset_vector                     17.7MB ± 0%    27.2MB ± 0%  +54.20%  (p=0.008 n=5+5)
+RangeQuery/at_modifier_                               28.3MB ± 0%    35.1MB ± 0%  +24.20%  (p=0.016 n=4+5)
+RangeQuery/at_modifier_with_positive_offset_vector    28.1MB ± 0%    35.0MB ± 0%  +24.36%  (p=0.008 n=5+5)
+RangeQuery/clamp                                      34.1MB ± 0%    29.2MB ± 0%  -14.36%  (p=0.008 n=5+5)
+RangeQuery/clamp_min                                  34.1MB ± 0%    29.1MB ± 0%  -14.41%  (p=0.008 n=5+5)
+RangeQuery/complex_func_query                         48.5MB ± 0%    31.1MB ± 0%  -35.99%  (p=0.008 n=5+5)
+RangeQuery/func_within_func_query                     35.0MB ± 0%    30.6MB ± 0%  -12.51%  (p=0.008 n=5+5)
+RangeQuery/aggr_within_func_query                     35.0MB ± 0%    30.6MB ± 0%  -12.51%  (p=0.008 n=5+5)
+RangeQuery/histogram_quantile                         49.4MB ± 0%    96.1MB ± 0%  +94.43%  (p=0.008 n=5+5)
```

Comment:
I wasn't able to track down where the excess memory originates from. I don't think it is that bad and maybe we can open an issue to track this?

Reply:
Yup makes sense!